### ISSUE TITLE
SC1004: This backslash+linefeed is literal. Command: sed -e '$a\ ...

### ISSUE THREAD

**Issue by HarriL** [koalaman/shellcheck#2326]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1004
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration, or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

FILE=/tmp/test.txt

cat <<-EOF >$FILE
first
second
third
last line
EOF

echo 'TEST1 Syntax error 1: after \ should be embedded newline'
sed -e '$a\1APPEND THIS TO BE THE LAST LINE' < "$FILE"

echo 'TEST2 Syntax error 2: after "a" should be \ and embedded newline'
sed -e '$a2APPEND THIS TO BE THE LAST LINE' < "$FILE"

echo "TEST3 Correct"
sed -e '$a\
APPEND THIS TO BE THE LAST LINE' < "$FILE"

```

#### Here's what shellcheck currently says:
```
Line 20	SC1004: This backslash+linefeed is literal. Break outside single quotes if you just want to break the line.
```
#### Here's what I wanted or expected to see:

1. TEST PROGRAM OUTPUT 
% sh test.sh

TEST1 Syntax error 1: after \ should be embedded newline
sed: 1: "$a\1APPEND THIS TO BE T ...": extra characters after \ at the end of a command
TEST2 Syntax error 2: after "a" should be \ and embedded newline
sed: 1: "$a2APPEND THIS TO BE TH ...": command a expects \ followed by text
TEST3 Correct
first
second
third
last line
APPEND THIS TO BE THE LAST LINE
% 

2.1 Expectation
SC1004 should not rise. The syntax of the TEST3 is correct!
Of course, all the tests are related to the sed command and not to the shell.

TODO: Describe expected/desired output
Can not guess b/c this is the first time I use this very useful shellcheck :-)

**Comment by koalaman** [comment]
This was a particularly ancient warning, and predated checks being verified on a corpus of scripts. While trying to improve the signal-to-noise ratio I found that it was almost all noise, so I retired it entirely. Thanks!
